src/textual/widgets/_input.py

```diff
@@ -330,6 +334,15 @@ def _watch_cursor_position(self) -> None:
 
         self.app.cursor_position = self.cursor_screen_offset
 
+    def _watch_cursor_blink(self, blink: bool) -> None:
+        """Ensure we handle updating the cursor blink at runtime."""
+        if self._blink_timer is not None:
+            if blink:
+                self._blink_timer.resume()
+            else:
+                self._cursor_visible = True
+                self._blink_timer.pause()
```

Comment:
The only reason the `if self._blink_timer is not None` guard exists is because this watcher can run before the `Input` is mounted, and the `_blink_timer` isn't initialised until `on_mount`.

I don't think we have to worry about the timer disappearing or anything.

Reply:
Exactly. So, what if the user sets cursor visibility before mounting the widget?
Then, it'll expect cursor visibility to have been set but it won't have been because we skipped the set altogether.

E.g., in a situation like this:

```py
def compose(self):
    inp = Input()
    inp.cursor_blink = False
    yield inp
```

Reply:
If they set the blink before the mount, then the blink timer will still be initialised in `on_mount`, but it will start as paused:

```python
    def _on_mount(self, _: Mount) -> None:
        self._blink_timer = self.set_interval(
            0.5,
            self._toggle_cursor,
            pause=not (self.cursor_blink and self.has_focus),
        )
```